test/org/sagebionetworks/bridge/models/schedules/CriteriaScheduleStrategyTest.java

```diff
@@ -0,0 +1,348 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.sagebionetworks.bridge.TestConstants.TEST_STUDY_IDENTIFIER;
+
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.sagebionetworks.bridge.TestConstants;
+import org.sagebionetworks.bridge.dynamodb.DynamoSchedulePlan;
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.json.BridgeObjectMapper;
+import org.sagebionetworks.bridge.models.ClientInfo;
+import org.sagebionetworks.bridge.models.accounts.User;
+import org.sagebionetworks.bridge.models.schedules.CriteriaScheduleStrategy.ScheduleCriteria;
+import org.sagebionetworks.bridge.validators.SchedulePlanValidator;
+import org.sagebionetworks.bridge.validators.Validate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.google.common.collect.Sets;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+public class CriteriaScheduleStrategyTest {
+    
+    private Schedule strategyWithAppVersions = makeValidSchedule("Strategy With App Versions");
+    private Schedule strategyWithOneRequiredDataGroup = makeValidSchedule("Strategy With One Required Data Group");
+    private Schedule strategyWithRequiredDataGroups = makeValidSchedule("Strategy With Required Data Groups");
+    private Schedule strategyWithOneProhibitedDataGroup = makeValidSchedule("Strategy With One Prohibited Data Group");
+    private Schedule strategyWithProhibitedDataGroups = makeValidSchedule("Strategy With One Prohibited Data Groups");
+    private Schedule strategyNoCriteria = makeValidSchedule("Strategy No Criteria");
```

Comment:
Minor: Can we make makeValidSchedule() static and all of these schedules static final?